src/AzureAuth/AuthModeExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth
+{
+    using Microsoft.Authentication.MSALWrapper;
+    using Microsoft.Office.Lasso.Interfaces;
+
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Extensions to <see cref="AuthMode"/>s.
+    /// </summary>
+    public static class AuthModeExtensions
+    {
+        /// <summary>
+        /// Filters a list of <see cref="AuthMode"/> for interaction returning a single aggregate <see cref="AuthMode"/>.
+        /// </summary>
+        /// <param name="authModes"><see cref="AuthMode"/>s to aggregate and filter.</param>
+        /// <param name="env">An <see cref="IEnv"/> to use.</param>
+        /// <returns>A single aggregate <see cref="AuthMode"/> filtered for interactivity.</returns>
+        public static AuthMode FilterInteraction(this IEnumerable<AuthMode> authModes, IEnv env)
+        {
+            if (InteractiveAuthDisabled(env))
+            {
+#if PlatformWindows
+                return AuthMode.IWA;
+#else
+                return 0;

```

Comment:
Basically here we are saying non-interactive auth in Mac or Linus is impossible?

Reply:
With public client auth, yes. That's correct. This code is also being lifted as-is from `CombinedAuthMode`, so this hasn't changed at all.

I did suggest to @kyle-rader-msft that this `0` be given a name like `None` though (better done in a separate PR).